pkg/kubelet/network/dns/dns_test.go

```diff
@@ -53,40 +53,48 @@ func TestParseResolvConf(t *testing.T) {
 		nameservers []string
 		searches    []string
 		options     []string
+		isErr       bool
 	}{
-		{"", []string{}, []string{}, []string{}},
-		{" ", []string{}, []string{}, []string{}},
-		{"\n", []string{}, []string{}, []string{}},
-		{"\t\n\t", []string{}, []string{}, []string{}},
-		{"#comment\n", []string{}, []string{}, []string{}},
-		{" #comment\n", []string{}, []string{}, []string{}},
-		{"#comment\n#comment", []string{}, []string{}, []string{}},
-		{"#comment\nnameserver", []string{}, []string{}, []string{}},
-		{"#comment\nnameserver\nsearch", []string{}, []string{}, []string{}},
-		{"nameserver 1.2.3.4", []string{"1.2.3.4"}, []string{}, []string{}},
-		{" nameserver 1.2.3.4", []string{"1.2.3.4"}, []string{}, []string{}},
-		{"\tnameserver 1.2.3.4", []string{"1.2.3.4"}, []string{}, []string{}},
-		{"nameserver\t1.2.3.4", []string{"1.2.3.4"}, []string{}, []string{}},
-		{"nameserver \t 1.2.3.4", []string{"1.2.3.4"}, []string{}, []string{}},
-		{"nameserver 1.2.3.4\nnameserver 5.6.7.8", []string{"1.2.3.4", "5.6.7.8"}, []string{}, []string{}},
-		{"nameserver 1.2.3.4 #comment", []string{"1.2.3.4"}, []string{}, []string{}},
-		{"search foo", []string{}, []string{"foo"}, []string{}},
-		{"search foo bar", []string{}, []string{"foo", "bar"}, []string{}},
-		{"search foo bar bat\n", []string{}, []string{"foo", "bar", "bat"}, []string{}},
-		{"search foo\nsearch bar", []string{}, []string{"bar"}, []string{}},
-		{"nameserver 1.2.3.4\nsearch foo bar", []string{"1.2.3.4"}, []string{"foo", "bar"}, []string{}},
-		{"nameserver 1.2.3.4\nsearch foo\nnameserver 5.6.7.8\nsearch bar", []string{"1.2.3.4", "5.6.7.8"}, []string{"bar"}, []string{}},
-		{"#comment\nnameserver 1.2.3.4\n#comment\nsearch foo\ncomment", []string{"1.2.3.4"}, []string{"foo"}, []string{}},
-		{"options ndots:5 attempts:2", []string{}, []string{}, []string{"ndots:5", "attempts:2"}},
-		{"options ndots:1\noptions ndots:5 attempts:3", []string{}, []string{}, []string{"ndots:5", "attempts:3"}},
-		{"nameserver 1.2.3.4\nsearch foo\nnameserver 5.6.7.8\nsearch bar\noptions ndots:5 attempts:4", []string{"1.2.3.4", "5.6.7.8"}, []string{"bar"}, []string{"ndots:5", "attempts:4"}},
+		{"", []string{}, []string{}, []string{}, false},
```

Comment:
since you are changing this part, it would be preferable to change all cases to use field names so that nil values (such as `[]string{}, false` can be omitted. The test cases will be much more readable.

Reply:
@bowei  `[]string{}` is empty not nil.  If I omit the fieldnames for those, they become nil and start failing.

Reply:
The only ones we can avoid are the bool and string variable.

Reply:
What does not work? Is it checking for nil value explicitly rather than iterating through the slice?

https://tour.golang.org/moretypes/12

Reply:
the function `parseResolvConf` returns empty slices not nil, so if i changed my  first two tests from 
```
  {"", []string{}, []string{}, []string{}, false},
  {" ", []string{}, []string{}, []string{}, false},
```
to 
```
{},
{data: " "},
```
it fails, because missing fields are interpreted as nil, while the function parseResolvConf is returning empty slices. I can change the function to return nil here that would require more changes. Let me know if understood your comment correctly ?

Reply:
Hmm.. this looks to be a limitation of testify EqualValues, as a nil value slice should be treated as equal:
```
	var a []string
	b := []string{}
	for i := range a {
		if b[i] != a[i] {
			t.Errorf("not equal")
		}
	}
```